File: tools/if3tool_url_query.cpp
#include "if3_world.hpp"
#include "if3_dynbuf.hpp"
#include "if3_tree.hpp"
#include "if3_compile.hpp"
#include "if3_kernel.hpp"

using namespace if3;

int main( int argc, char **argv )
{
  const char *IF3_home = getenv("IF3_HOME");
  if( !IF3_home )
    IF3_home = "/opt/IF3/";
  filename_t home = IF3_home;
  filename_t sharedir = home + "share/if3db-pre/";
  filename_t compiled_hostname_file = sharedir + "hostnames.pre";
  filename_t compiled_url_file = sharedir + "urls.pre";
  filename_t compiled_phrase_file = sharedir + "phrases.pre";

  const char *hostname = argv[1];
  size_t hostname_length = strlen(hostname);

  const char *link = argv[1];
  size_t link_length = strlen(link);

  bool enable_proof=true;

  kernel::kernel_t *k = new kernel::kernel_t(
                                             compiled_hostname_file,
                                             compiled_url_file,
                                             compiled_phrase_file
                                             );

  kernel::categories_enable_t categories_enable_bits;

  kernel::full_scan_request_t request;
  kernel::full_scan_results_t results;

  request.hostname( hostname, hostname_length );
  request.link( link, link_length );
  int category=-1;

  category = k->perform_scan(results,request,categories_enable_bits,enable_proof);

  std::stringstream ss;
  ss << results << "\n";

  std::cout << "category: " << category << "\n";
  std::cout << results;

  delete k;
  return 0;
}

File: tools/if3tool_url_query.py
#!/usr/bin/env python

import os
import sys
import if3kernel
import urlparse


def main():
  IF3_home = '/opt/IF3'
  if os.environ.has_key('IF3_HOME'):
    IF3_home = os.environ['IF3_HOME']
  precompiled_path = os.path.join( IF3_home, 'share/if3db-pre' )
  if3kernel.startup( 
    os.path.join(precompiled_path,'hostnames.pre'),
    os.path.join(precompiled_path,'urls.pre'),
    os.path.join(precompiled_path,'phrases.pre')
    )
  
  url = urlparse.urlsplit( sys.argv[1] )
  
  print if3kernel.scan_url( 1, url.netloc, url.geturl() )

if __name__ == "__main__":
    main()
    if3kernel.shutdown()
    

